src/ServiceManagement/Compute/Commands.ServiceManagement.Test/FunctionalTests/ServiceManagementCmdletTestHelper.cs

```diff
@@ -1200,17 +1199,17 @@ public ExtensionConfigurationInput NewAzureServiceDiagnosticsExtensionConfig
 
         // Set-AzureServiceDiagnosticsExtension
         public ManagementOperationContext SetAzureServiceDiagnosticsExtension
-            (string service, AzureStorageContext storageContext, string config = null, string[] roles = null, string slot = null)
+            (string service, IStorageContext storageContext, string config = null, string[] roles = null, string slot = null)
         {
             return RunPSCmdletAndReturnFirst<ManagementOperationContext>(new SetAzureServiceDiagnosticsExtensionCmdletInfo(service, storageContext, config, roles, slot));
         }
 
-        public ManagementOperationContext SetAzureServiceDiagnosticsExtension(string service, AzureStorageContext storageContext, X509Certificate2 cert, string config = null, string[] roles = null, string slot = null)
+        public ManagementOperationContext SetAzureServiceDiagnosticsExtension(string service, IStorageContext storageContext, X509Certificate2 cert, string config = null, string[] roles = null, string slot = null)
         {
             return RunPSCmdletAndReturnFirst<ManagementOperationContext>(new SetAzureServiceDiagnosticsExtensionCmdletInfo(service, storageContext, cert, config, roles, slot));
```

Comment:
Doesn't this generic method need to be changed to support IStorageContext?

Reply:
It doesn't need to be changed. Just want to clean up the places where it doesn't need to be a specific storage type.
I'll keep it like this unless you have strong preference.